crates/node/src/command.rs

```diff
@@ -55,71 +55,97 @@ impl SubstrateCli for Cli {
     }
 }
 
-/// Parse and run command line arguments
-pub fn run() -> sc_cli::Result<()> {
-    let mut cli = Cli::from_args();
+fn set_dev_environment(cli: &mut Cli) {
+    // create a reproducible dev environment
+    cli.run.run_cmd.shared_params.dev = false;
+    cli.run.run_cmd.shared_params.chain = Some("dev".to_string());
 
-    // alias madara_path <> base_path
-    // TODO also alias tmp (tmp generates random base_paths that are not specified within
-    // the command)
-    let madara_path = match (cli.run.madara_path.clone(), cli.run.run_cmd.shared_params.base_path.clone()) {
-        (Some(madara_path), _) => {
-            cli.run.run_cmd.shared_params.base_path = Some(madara_path.clone());
-            madara_path.to_str().unwrap().to_string()
-        }
-        (_, Some(base_path)) => {
-            cli.run.madara_path = Some(base_path.clone());
-            base_path.to_str().unwrap().to_string()
-        }
-        _ => {
-            let home_path = std::env::var("HOME").unwrap_or(std::env::var("USERPROFILE").unwrap_or(".".into()));
-            let path = format!("{}/.madara", home_path);
-            cli.run.run_cmd.shared_params.base_path = Some((path.clone()).into());
-            cli.run.madara_path = Some((path.clone()).into());
-            path
-        }
-    };
-
-    if let Some(genesis_url) = cli.run.genesis_url.clone() {
-        // can't copy extra genesis-assets atm
-        // we can reuse #982 to create the standard to fetch relevant files
-        utils::fetch_from_url(genesis_url, madara_path.clone() + "/configs/genesis-assets")?;
-    } else {
-        // TODO confirm with the CI that we are fetching all and fetch dynamically
-        // Issue #982
-        for file in constants::GENESIS_ASSETS_FILES {
-            let src_path = utils::get_project_path();
-            if let Ok(src_path) = src_path {
-                let src_path = src_path + "/configs/genesis-assets/" + file;
-                utils::copy_from_filesystem(src_path, madara_path.clone() + "/genesis-assets")?;
-            } else {
-                utils::fetch_from_url(
-                    constants::GENESIS_ASSETS_URL.to_string() + file,
-                    madara_path.clone() + "/genesis-assets",
-                )?;
-            }
-        }
-    }
+    cli.run.run_cmd.force_authoring = true;
+    cli.run.run_cmd.alice = true;
+
+    // we can't set `--rpc-cors=all`, so it needs to be set manually if we want to connect with external
+    // hosts
+    cli.run.run_cmd.rpc_external = true;
+    cli.run.run_cmd.rpc_methods = RpcMethods::Unsafe;
+}
+
+fn set_testnet(cli: &mut Cli, testnet: Testnet) -> Result<(), String> {
+    // checks if it should retrieve and enable a specific chain-spec
+    let madara_path = cli.run.madara_path.clone().expect("Failed retrieving madara_path").to_str().unwrap().to_string();
```

Comment:
Why is it safe to unwrap?
Why can we `expect` it?

Reply:
Also recommended message  format for `expect` is defined here: https://doc.rust-lang.org/std/result/enum.Result.html#recommended-message-style

Reply:
because the value is always a string (is it not?)

Reply:
applied the `into_os_string` approach with an `expect` error msg. Seems cleaner now